Behavior difference: tsgo no longer infer variadic `args` type when `arguments[...]` is referenced inside plain js function body

There are (somewhat) related sections on `CHANGES.md`, but none of them are describing what I observe here.
- https://github.com/microsoft/typescript-go/blob/main/CHANGES.md#jsdoc-variadic-types-are-now-only-synonyms-for-array-types
- https://github.com/microsoft/typescript-go/blob/main/CHANGES.md#with-strict-false-corsa-no-longer-allows-omitting-arguments-for-parameters-with-type-undefined-unknown-or-any

---

## Steps to reproduce

1. Start a plain new project
    - "@typescript/native-preview": "7.0.0-dev.20260606.1",
    - "typescript": "6.0.3"
2. Run `npx tsgo --init` to get the default tsconfig. Turn on these flags as well.
    - "allowJs": true,
    - "checkJs": true,
    - "stableTypeOrdering": true,
3. Add `src/main.js`. Run `npx tsc` & `npx tsgo` to see the difference.

```js
// src/main.js

/**
 * @param {number} length
 */
export function rectArea(length) {
    const width = arguments[1];

    if (typeof width === 'number') {
        return length * width;
    }

    return length * length;
}

rectArea(3);
rectArea(4, 5);
```

## Behavior with `typescript@6.0`

Emit

```ts
/**
 * @param {number} length
 */
export function rectArea(length: number, ...args: any[]): number;
//# sourceMappingURL=main.d.ts.map
```

With no error.

## Behavior with `tsgo`

Emit

```ts
/**
 * @param {number} length
 */
export declare function rectArea(length: number): number;
//# sourceMappingURL=main.d.ts.map
```

With error.

```
src/main.js:17:13 - error TS2554: Expected 1 arguments, but got 2.

17 rectArea(4, 5);
               ~


Found 1 error in src/main.js:17
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -187,6 +187,22 @@ const FORWARD = 1,
   BACKWARD = 2;
 ```
 
+#### `arguments` usage within a JS function body no longer implies a `...args: any[]` parameter
+```js
+function f() {
+  return arguments[0];
+}
+f("something"); // Previously allowed, now an error
+```
+
+Instead, explicitly use a spread parameter:
+```js
+function f(...args) {
+  return args[0];
+}
+f("something"); // Still allowed
+```
+
 ### JSDoc Types
 
 #### JSDoc variadic types are now only synonyms for array types.

